### ISSUE TITLE
跑unsloth版本会出现维度不匹配的问题

### ISSUE THREAD

**Issue by wangeurak** [datawhalechina/unlock-deepseek#38]
好奇怪，跑了三个epoch就会出现维度问题的报错，我是在单机单卡4090的环境下跑的，之前跑非lora版本是不会出现维度问题的，想请教一下大佬这是什么原因
报错信息

swanlab: 🚀 View run at https://swanlab.cn/@wangyuxin/Datawhale-R1-by_wyx/runs/7fj14medpnmkpz8zu6bhd
{'loss': -0.0, 'grad_norm': 0.09001431614160538, 'learning_rate': 7.142857142857143e-07, 'rewards/format_reward_func': 0.25, 'rewards/equation_reward_func': 0.0, 'reward': 0.25, 'reward_std': 0.5, 'completion_length': 445.0625, 'kl': 0.0, 'epoch': 0.0}                          
{'loss': 0.0, 'grad_norm': 0.0318080298602581, 'learning_rate': 1.4285714285714286e-06, 'rewards/format_reward_func': 0.0625, 'rewards/equation_reward_func': 0.0, 'reward': 0.0625, 'reward_std': 0.125, 'completion_length': 438.9375, 'kl': 0.0, 'epoch': 0.0}                     
{'loss': 0.0, 'grad_norm': 0.07568195462226868, 'learning_rate': 2.1428571428571427e-06, 'rewards/format_reward_func': 0.1875, 'rewards/equation_reward_func': 0.0625, 'reward': 0.25, 'reward_std': 0.39433756470680237, 'completion_length': 428.9375, 'kl': 0.0008076233643805608, 'epoch': 0.0}
  1%|█▌                                                                                                                                                                                                                                             | 3/450 [01:38<3:59:59, 32.21s/it]Unsloth: Input IDs of length 1025 > the model's max sequence length of 1024.
We shall truncate it ourselves. It's imperative if you correct this issue first.
swanlab: Error happened while training
swanlab: 🌟 Run `swanlab watch /autodl-fs/data/unlock-deepseek/Datawhale-R1/swanlog` to view SwanLab Experiment Dashboard locally
swanlab: 🏠 View project at https://swanlab.cn/@wangyuxin/Datawhale-R1-by_wyx
swanlab: 🚀 View run at https://swanlab.cn/@wangyuxin/Datawhale-R1-by_wyx/runs/7fj14medpnmkpz8zu6bhd
  File "/autodl-fs/data/unlock-deepseek/Datawhale-R1/train_Datawhale-R1_unsloth.py", line 354, in <module>
    main()
  File "/autodl-fs/data/unlock-deepseek/Datawhale-R1/train_Datawhale-R1_unsloth.py", line 351, in main
    grpo_function(model_args, dataset_args, training_args, callbacks=callbacks)
  File "/autodl-fs/data/unlock-deepseek/Datawhale-R1/train_Datawhale-R1_unsloth.py", line 293, in grpo_function
    train_result = trainer.train()
                   ^^^^^^^^^^^^^^^
  File "/root/miniconda3/lib/python3.12/site-packages/transformers/trainer.py", line 2171, in train
    return inner_training_loop(
           ^^^^^^^^^^^^^^^^^^^^
  File "<string>", line 382, in _fast_inner_training_loop
  File "<string>", line 25, in _unsloth_training_step
  File "/autodl-fs/data/unlock-deepseek/Datawhale-R1/unsloth_compiled_cache/UnslothGRPOTrainer.py", line 877, in _prepare_inputs
    ref_per_token_logps = self._get_per_token_logps(
                          ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/autodl-fs/data/unlock-deepseek/Datawhale-R1/unsloth_compiled_cache/UnslothGRPOTrainer.py", line 792, in _get_per_token_logps
    logits = model(input_ids=input_ids, attention_mask=attention_mask, logits_to_keep=logits_to_keep + 1).logits
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/root/miniconda3/lib/python3.12/site-packages/torch/nn/modules/module.py", line 1736, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/root/miniconda3/lib/python3.12/site-packages/torch/nn/modules/module.py", line 1747, in _call_impl
    return forward_call(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/root/miniconda3/lib/python3.12/site-packages/unsloth/models/llama.py", line 1195, in PeftModelForCausalLM_fast_forward
    return self.base_model(
           ^^^^^^^^^^^^^^^^
  File "/root/miniconda3/lib/python3.12/site-packages/torch/nn/modules/module.py", line 1736, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/root/miniconda3/lib/python3.12/site-packages/torch/nn/modules/module.py", line 1747, in _call_impl
    return forward_call(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/root/miniconda3/lib/python3.12/site-packages/peft/tuners/tuners_utils.py", line 197, in forward
    return self.model.forward(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/root/miniconda3/lib/python3.12/site-packages/unsloth/models/llama.py", line 1055, in _CausalLM_fast_forward
    outputs = self.model(
              ^^^^^^^^^^^
  File "/root/miniconda3/lib/python3.12/site-packages/torch/nn/modules/module.py", line 1736, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/root/miniconda3/lib/python3.12/site-packages/torch/nn/modules/module.py", line 1747, in _call_impl
    return forward_call(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/root/miniconda3/lib/python3.12/site-packages/unsloth/models/llama.py", line 703, in LlamaModel_fast_forward
    inputs_embeds *= attention_mask.unsqueeze(0).transpose(0, 1).transpose(1, 2)
The size of tensor a (1024) must match the size of tensor b (1025) at non-singleton dimension 1
wandb: 
wandb: You can sync this run to the cloud by running:
wandb: wandb sync /autodl-fs/data/unlock-deepseek/Datawhale-R1/wandb/offline-run-20250218_202031-phnmsvyd
wandb: Find logs at: ../../../../autodl-fs/data/unlock-deepseek/Datawhale-R1/wandb/offline-run-20250218_202031-phnmsvyd/logs
  1%|█▌                                                                                                                                                                                                                                             | 3/450 [02:11<5:27:10, 43.92s/it]
[rank0]:[W218 20:22:48.481796043 ProcessGroupNCCL.cpp:1250] Warning: WARNING: process group has NOT been destroyed before we destruct ProcessGroupNCCL. On normal program exit, the application should call destroy_process_group to ensure that any pending NCCL operations have finished in this process. In rare cases this process can exit before this point and block the progress of another member of the process group. This constraint has always been present,  but this warning has only been added since PyTorch 2.4 (function operator())
